pytools/__init__.py

```diff
@@ -2182,7 +2182,7 @@ def __init__(  # pylint: disable=too-many-arguments
         # feature for interactive use, do not do it unless there is (weak)
         # evidence of interactive use.
         import sys
-        use_late_start_logging = sys.stdin.isatty()
+        use_late_start_logging = bool(sys.stdin and sys.stdin.isatty())
```

Comment:
This is super hard to read IMO---or easy to read incorrectly. Just spell it out:
```py
if sys.stdin is None:
    # Can happen, e.g., if pudb is controlling the console.
    use_late_start_logging = False
else sys.stdin.isatty():
    use_late_start_logging = sys.stdin.isatty()
```

Reply:
OK, I have replaced the line.